### PR TITLE
Music releases project - Mats Johansen & Gabriella Bolin

### PR DESCRIPTION
**Pull request by Gabbi-89**
Pair-programming with @mats-johansen 

Link to our deployed project:
https://musicreleases.netlify.app/

Our reflections on how this week's project turned out:
The pair-programming went very well, except from technical issues, probably due to trying to add more than one element per 'return'. 
Struggled most with getting the buttons correctly over the album-image when hovering and adding the comma between the artist.

### CODE REVIEW

File: code/src/Components/AlbumDiv.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                    <a 
+                        key={artist.external_urls.spotify} 
+                        href={artist.external_urls.spotify} 
+                        target="_blank" 
+                        rel="noopener noreferrer" 
+                        className="artists-links">
+                        <span key={artist.name} className="artist-name">{artist.name}</span>
+                    </a>
+                    )
+                })}
```
Comment (HIPPIEKICK) [inline code/src/Components/AlbumDiv.js:26]: Consider doing this in another component and naming the prop artists:
`artists={album.artists}`
E.g. in the `<AlbumInfo/>` component, you could mount an `<Artist/>` component:

```
{props.artists.map(item => {
  return (
    <Artist
      key={item.id}
      item={item}
      artistLink={artist.external_urls.spotify} 
      className="artists-links">/>
  )
})}
```
and then have the actual a tag in the `<Artist/>` component

File: code/src/Components/AlbumInfo.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { Icons } from "./Icons";
+
+// The album-card containing the props and including the icons. 
+export const AlbumInfo = props => {
+  return (
+  <section className="album-card">
```
Comment (HIPPIEKICK) [inline code/src/Components/AlbumInfo.js:7]: Indentation:
```
return (
  <section className="album-card">
```

File: code/src/index.css [hunk 3]
```diff
@@ -1,13 +1,164 @@
...
+  /* Adds a comma and &-sign on right place + styling of it  */
+    .artists-links:not(:last-child):not(:nth-last-child(2))::after {
+    content: ', ';
+    color: #a0a0a0;
+  }
+  
+  .artists-links:not(:first-child):last-child::before {
+    content: ' & ';
+    color: #a0a0a0;
+  }
```
Comment (HIPPIEKICK) [inline code/src/index.css:82]: ⭐

File: code/src/index.css [hunk 4]
```diff
@@ -1,13 +1,164 @@
...
+      /*Wrapping on rows instead*/
+      flex-direction: row;
+      flex-wrap: wrap;
+    }
+
+    .album-card {
+      width: 40%;
+      height: 350px;
+    }
+  }
```
Comment (HIPPIEKICK) [inline code/src/index.css:158]: When the screen is around 1000px wide:

Skärmavbild 2020-10-23 kl  17 10 07

File: code/src/App.js [hunk 5]
```diff
@@ -1,12 +1,16 @@
...
+  return ( 
+    <section className="site-container">
+      
+        <Header />
+        <AlbumContainer />
+      
+    </section>
   )
-}
+}
```
Comment (HIPPIEKICK) [inline code/src/App.js:16]: semicolons?

**Review by HIPPIEKICK — CHANGES_REQUESTED** [review]
Good job with this project! Super nice hover effects 🥳 Please just check so that you're consistent regarding indentation and semicolons and I'll approve 👌

File: code/src/Components/AlbumCard.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+// The actual div containing all information such as image, artist name, album name and the correct URL-links
+export const AlbumCard = () => {
+  return (
+    <>
+      {data.albums.items.map(album => { // Mapping through the album array to get album-name, images, and the artist array.
+        return <AlbumInfo 
+          key={album.name}
+          image={album.images[0].url} 
+          name={album.name} 
+          artist={album.artists.map(artist => { // Mapping through the artist array for when having several artists
```
Comment (HIPPIEKICK) [inline code/src/Components/AlbumCard.js:15]: When our return statement is on several lines, we should wrap it in parentheses:
```
return (
  //code
  //code
)
```

File: code/src/Components/AlbumCard.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+            return ( 
+              <a 
+                key={artist.external_urls.spotify} 
+                href={artist.external_urls.spotify} 
+                target="_blank" 
+                rel="noopener noreferrer" 
+                className="artists-links">
+                <span key={artist.name} className="artist-name">{artist.name}</span>
+              </a>
+            )
```
Comment (HIPPIEKICK) [inline code/src/Components/AlbumCard.js:25]: Semicolon

File: code/src/Components/AlbumContainer.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+import { AlbumCard } from './AlbumCard';
+
+// Container for all the album-cards
+export const AlbumContainer = () => {
+  return (
+  <section className="album-container">
+    <AlbumCard /> 
+  </section>
```
Comment (HIPPIEKICK) [inline code/src/Components/AlbumContainer.js:10]: indent

File: code/src/index.css [hunk 9]
```diff
@@ -1,13 +1,164 @@
...
-    sans-serif;
-  -webkit-font-smoothing: antialiased;
-  -moz-osx-font-smoothing: grayscale;
-}
-
-code {
-  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
-    monospace;
-}
+  /*** GENERAL STYLING ***/
```
Comment (HIPPIEKICK) [inline code/src/index.css:1]: Everything in this file is indented for some reason

Comment (Gabbi-89) [reply code/src/index.css:1]: OMG! Thanks for noticing! Solved! :)

**Review by HIPPIEKICK — APPROVED** [review]

**Pull request closed without merging**